### PR TITLE
Add metrics

### PR DESCRIPTION
**Pull request by alonkashtan**

### CODE REVIEW

File: api.go [hunk 1]
```diff
@@ -108,6 +112,49 @@ func (server *ApiServer) ProxyIndex(response http.ResponseWriter, request *http.
 	}
 }
 
+func (server *ApiServer) GetMetricEvents(response http.ResponseWriter, request *http.Request) {
+	token := request.URL.Query().Get("token")
```
Comment (danielshalev) [inline api.go]: maybe call this query param offset instead of token as i wasn't sure what's it for at first sight

File: link.go [hunk 2]
```diff
@@ -45,23 +48,40 @@ func NewToxicLink(proxy *Proxy, collection *ToxicCollection, direction stream.Di
...
 		if err != nil {
 			logrus.WithFields(logrus.Fields{
-				"name":  link.proxy.Name,
+				"name":  proxyName,
 				"bytes": bytes,
 				"err":   err,
 			}).Warn("Source terminated")
+
+			metrics.RegisterEvent(metrics.Event{
+				EventType: "Client Disconnected",
```
Comment (danielshalev) [inline link.go]: Is there a way to see all event types? maybe should be constants

File: proxy.go [hunk 3]
```diff
@@ -167,6 +170,7 @@ func (proxy *Proxy) server() {
 				"proxy":    proxy.Listen,
 				"upstream": proxy.Upstream,
 			}).Error("Unable to open connection to upstream")
+			metrics.RegisterEvent(metrics.Event{EventType: "Upstream unavailable", Client: client.RemoteAddr().String(), Upstream: proxy.Upstream, ProxyName: proxy.Name, Time: time.Now()})
```
Comment (danielshalev) [inline proxy.go]: use upper case U in unavailable

**Review by danielshalev — APPROVED** [review]

File: api.go (hunk 1 above)
Comment (alonkashtan) [reply api.go]: I see your problem. "Offset" is problematic as it is a string, with no promise about it's structure. I currently pass there a number, but in past projects came up a need to add information to the string later (for example for fault-tolerance scenarios). So I prefer to give a more opaque name.
Maybe make the field in the response named `location` and the query param `afterLocation`. What do you think?

Comment (danielshalev) [reply api.go]: sounds good

**Pull request merged**